rbpf-cli/src/main.rs

```diff
@@ -0,0 +1,226 @@
+use clap::{App, Arg};
+use serde::{Deserialize, Serialize};
+use serde_json::Result;
+use solana_bpf_loader_program::{
+    BpfError,
+    create_vm,
+    serialization::serialize_parameters,
+    syscalls::register_syscalls,
+    ThisInstructionMeter,
+};
+use solana_rbpf::{
+    assembler::assemble,
+    static_analysis::Analysis,
+    verifier::check,
+    vm::{
+        Config, DynamicAnalysis, Executable,
+    },
+};
+use solana_sdk::{
+    account::AccountSharedData,
+    bpf_loader,
+    keyed_account::KeyedAccount,
+    process_instruction::{InvokeContext, MockInvokeContext},
+    pubkey::Pubkey,
+};
+use std::{cell::RefCell, fs::File, io::Read, path::Path};
+
+#[derive(Serialize, Deserialize)]
+struct Data {
+    accounts: Vec<AccountSharedData>,
+    insndata: Vec<u8>,
+}
+fn load_accounts<'a>(path: &Path) -> Result<Data> {
+    let file = File::open(path).unwrap();
+    let input: Data = serde_json::from_reader(file)?;
+    println!("accounts {:?}", &input.accounts);
+    println!("insndata {:?}", &input.insndata);
+    Ok(input)
+}
+
+fn main() {
+    solana_logger::setup();
+    let matches = App::new("Solana BPF CLI")
+        .version("0.2.8")
+        .author("Solana Maintainers <maintainers@solana.foundation>")
+        .about("CLI to test and analyze eBPF programs")
+        .arg(
+            Arg::new("assembler")
+                .about("Assemble and load eBPF executable")
+                .short('a')
+                .long("asm")
+                .value_name("FILE")
+                .takes_value(true)
+                .required_unless_present("elf"),
+        )
+        .arg(
+            Arg::new("elf")
+                .about("Load ELF as eBPF executable")
+                .short('e')
+                .long("elf")
+                .value_name("FILE")
+                .takes_value(true)
+                .required_unless_present("assembler"),
+        )
+        .arg(
+            Arg::new("input")
+                .about("Input for the program to run on")
+                .short('i')
+                .long("input")
+                .value_name("FILE / BYTES")
+                .takes_value(true)
+                .default_value("0"),
+        )
+        .arg(
+            Arg::new("memory")
+                .about("Heap memory for the program to run on")
+                .short('m')
+                .long("mem")
+                .value_name("BYTES")
+                .takes_value(true)
+                .default_value("0"),
+        )
+        .arg(
+            Arg::new("use")
```

Comment:
Some description of these would be nice, for example, what is "cfg"?

Reply:
I'll add descriptions.